server/remote_cache/digest/digest.go

```diff
@@ -583,22 +583,23 @@ func Diff(s1 []*repb.Digest, s2 []*repb.Digest) (missingFromS1 []*repb.Digest, m
 	missingFromS1 = make([]*repb.Digest, 0)
 	missingFromS2 = make([]*repb.Digest, 0)
 
-	s1Set := make(map[*repb.Digest]struct{}, len(s1))
+	s1Set := make(map[Key]*repb.Digest, len(s1))
```

Comment:
It seems redundant to store both a Key and a *Digest which contain the same data. What do you think about calling key.ToDigest() when populating the slices at the end?

Reply:
Sure, it's either redundant map values or redundant conversion back into digest protos. I guessed storing pointer values in the map would be slightly more performant than re-allocating new Keys for each set of differences, but probably either way is fine. Changed.